.github/workflows/sanitizers.yml

```diff
@@ -0,0 +1,170 @@
+        -DUMF_BUILD_SHARED_LIBRARY=${{matrix.shared_library}}
+        -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+        -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+        -DUMF_BUILD_OS_MEMORY_PROVIDER=${{matrix.os_provider}}
+        -DUMF_ENABLE_POOL_TRACKING=${{matrix.pool_tracking}}
+        -DUMF_FORMAT_CODE_STYLE=OFF
+        -DUMF_DEVELOPER_MODE=ON
+        -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=ON
+        -DUMF_BUILD_LIBUMF_POOL_DISJOINT=ON
+        -DUMF_BUILD_LIBUMF_POOL_SCALABLE=ON
+        -DUSE_ASAN=${{matrix.sanitizers.asan}}
+        -DUSE_UBSAN=${{matrix.sanitizers.ubsan}}
+        -DUSE_TSAN=${{matrix.sanitizers.tsan}}
+        -DUMF_BUILD_EXAMPLES=ON
+
+    - name: Build UMF
+      run: cmake --build ${{env.BUILD_DIR}} -j $(nproc)
+
+    - name: Run tests
+      working-directory: ${{env.BUILD_DIR}}
+      run: ctest --output-on-failure
+
+  ubuntu-build:
+    name: Ubuntu
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04']
+        build_type: [Debug]
+        compiler: [{c: gcc, cxx: g++}, {c: clang, cxx: clang++}]
+        pool_tracking: ['ON']
+        shared_library: ['OFF']
+        os_provider: ['ON']
+        # TSAN is mutually exclusive with other sanitizers
+        sanitizers: [{asan: ON, ubsan: ON, tsan: OFF}, {asan: OFF, ubsan: OFF, tsan: ON}]
+    runs-on: ${{matrix.os}}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Install apt packages
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y clang cmake libhwloc-dev libnuma-dev libjemalloc-dev libtbb-dev
+
+    - name: Install g++-7
+      if: matrix.compiler.cxx == 'g++-7'
+      run: sudo apt-get install -y ${{matrix.compiler.cxx}}
+
+    - name: Configure build
+      run: >
+        cmake
+        -B ${{env.BUILD_DIR}}
+        -DCMAKE_BUILD_TYPE=${{matrix.build_type}}
+        -DUMF_BUILD_SHARED_LIBRARY=${{matrix.shared_library}}
+        -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+        -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+        -DUMF_BUILD_OS_MEMORY_PROVIDER=${{matrix.os_provider}}
+        -DUMF_ENABLE_POOL_TRACKING=${{matrix.pool_tracking}}
+        -DUMF_FORMAT_CODE_STYLE=OFF
+        -DUMF_DEVELOPER_MODE=ON
+        -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=ON
+        -DUMF_BUILD_LIBUMF_POOL_DISJOINT=ON
+        -DUMF_BUILD_LIBUMF_POOL_SCALABLE=ON
+        -DUSE_ASAN=${{matrix.sanitizers.asan}}
+        -DUSE_UBSAN=${{matrix.sanitizers.ubsan}}
+        -DUSE_TSAN=${{matrix.sanitizers.tsan}}
+        -DUMF_BUILD_EXAMPLES=ON
+
+    - name: Build UMF
+      run: cmake --build ${{env.BUILD_DIR}} -j $(nproc)
+
+    - name: Run tests
+      working-directory: ${{env.BUILD_DIR}}
+      run: ctest --output-on-failure
+
+  windows-build:
+    name: Windows
+    strategy:
+      matrix:
+        os: ['windows-2022']
+        build_type: [Debug]
+        compiler: [{c: cl, cxx: cl}, {c: clang-cl, cxx: clang-cl}]
+        pool_tracking: ['OFF']
```

Comment:
Just hardcode variables which have only one value allowed, it will be easier to read the file. Just like you do for several other CMake config options:
```
        -DUMF_FORMAT_CODE_STYLE=OFF
        -DUMF_DEVELOPER_MODE=ON
        -DUMF_BUILD_LIBUMF_POOL_DISJOINT=ON
```